### DISCUSSION TITLE
NetBeans 12 32 Bit

### DISCUSSION THREAD

**Discussion by jmccay-work** [apache/netbeans#5364] [Q&A] [answered]
The company I work for is currently stuck on Java 8 32 bit because of external dependencies that can't be changed currently.  I was looking on the website for the 32 bit downloads.  I currently am running the following on my system:

`openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
OpenJDK Server VM (build 25.292-b10, mixed mode)` 

The NetBeans information is:
`Product Version: Apache NetBeans IDE 12.0
Java: 1.8.0_292; OpenJDK Server VM 25.292-b10
Runtime: OpenJDK Runtime Environment 1.8.0_292-b10
System: Windows 10 version 10.0 running on x86; Cp1252; en_US (nb)`

Is there someplace I can download the 32-bit versions of NetBeans 12?

**Comment by lkishalmi** [comment]
Well, it's https://netbeans.apache.org/download/nb120/nb120.html Though you need to download the binary zip. That works with 32 bit JVM-s as well. We never released a 32 bit installer.

You might consider using NetBeans 12.6 as that is the last official build Supports Java 8 as runtime.
However if you can upgrade on the runtime you can even use the upcoming NetBeans 17, the binary zip still supports 32 bit Windows (and Linux execution), and it supports project written in Java 8.

**Comment by jmccay-work** [comment]
Thanks for the answer.  Unfortunately, I can't upgrade past Java 8.  The company I work for has a lot of products developed in Java 8 32-bit.  We plan to upgrade in the future, but right now we have a push to another design format for our products to minimize the use of Java on the client systems and relegating the use of Java to the server only.

I tried to install NetBeans 12.4 & 12.5.  According to the download website, Netbeans 12.6 requires JDK 11+.  Netbeans 12.4 & 12.5 didn't find my OpenJDK Java 8 32-bit.  It kept telling me that it needed a version of Java.

**Comment by neilcsmith-net** [comment, marked as answer]
@jmccay-work I suggest you separate out the runtime JDK for the IDE from the JDK you need for projects.  You can register alternative JDKs to work with (eg. your 32-bit Java 8) under `Tools / Java Platforms`.  I run the IDE on JDK 19 but still build and run some projects on JDK 8.

If it helps to have a contained installation that runs on a local JDK, then check out the community installers - linked from main Apache NetBeans download pages and distributed by my company - https://www.codelerity.com/netbeans/

> You might consider using NetBeans 12.6 as that is the last official build Supports Java 8 as runtime.
...
> According to the download website, Netbeans 12.6 requires JDK 11+.

NetBeans 12.6 only supports running on JDK 8 if built from source.  The Apache binaries only support JDK 11+.  That wasn't the plan when preparing the release, but some issues were found.

**Reply by lkishalmi** [reply]
Many of us does the IDE Runtime JDK and the Project Runtime JDK Separation as Neil mentions. I personally running NetBeans on JDK 17, and have Java 8, Java 11 projects, though sometimes I do tests with Java 17 and Java 19 as well.

**Comment by jmccay-work** [comment]
@neilcsmith-net and @lkishalmi, this sounds like the best idea for now.  I will look into doing that with my system.  Thanks for all your comments and help.
